src/components/WeeklyCalendar/index.js

```diff
@@ -0,0 +1,148 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useLocale } from '../../libs/hooks';
+import { useCurrentWeek, useYearRange, useFormattedWeek, useDisabledControls } from './hooks';
+import { addWeeks } from './helpers';
+import Select from './../Select';
+import RightIcon from './icons/rightArrow';
+import LeftIcon from './icons/leftArrow';
+import Header from './header';
+import Week from './week';
+import Hours from './hours';
+import StyledContainer from './styled/container';
+import StyledContent from './styled/content';
+import StyledControls from './styled/controls';
+import StyledTitle from './styled/title';
+import StyledArrowButton from './styled/arrowButton';
+import StyledGradient from './styled/gradient';
+
+export default function WeeklyCalendar(props) {
+    const {
+        events,
+        currentWeek,
+        minDate,
+        maxDate,
+        locale: localLocale,
+        onWeekChange,
+        onEventClick,
+        className,
+        style,
+    } = props;
+    const locale = useLocale(localLocale);
+    const hoursRef = useRef();
+    const daysRef = useRef();
+    const week = useCurrentWeek(currentWeek);
+    const formattedWeek = useFormattedWeek(week, locale);
+    const [today, setToday] = useState(new Date());
+    const yearsRange = useYearRange(minDate, maxDate, week);
+    const { disableNext, disablePrevious } = useDisabledControls(
```

Comment:
`disableNext`, `disablePrevious` are boolean variables, rename them to `isDisableNext`, `isDisablePrevious`